runtime/drivers/clickhouse/olap.go

```diff
@@ -251,7 +251,94 @@ func (c *connection) InsertTableAsSelect(ctx context.Context, name, sql string,
 		})
 	}
 
-	// merge strategy is also not supported for clickhouse
+	if strategy == drivers.IncrementalStrategyMerge {
+		_, onCluster, err := informationSchema{c: c}.entityType(ctx, "", name)
+		if err != nil {
+			return err
+		}
+		onClusterClause := ""
+		if onCluster {
+			onClusterClause = "ON CLUSTER " + safeSQLName(c.config.Cluster)
+		}
+		// Get the engine info of the given table
+		var engine string
+		res, err := c.Execute(ctx, &drivers.Statement{
+			Query:    "SELECT engine FROM system.tables WHERE database = currentDatabase() AND name = ?",
+			Args:     []any{name},
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		defer res.Close()
```

Comment:
May want to release this connection early. Two ideas :
1. Close this after scanning and if scan fails.
2. Create a simple utility function to get `engine` where the `defer` construct can be used.